frontend/src/components/Form.js

```diff
@@ -1,55 +1,85 @@
-import React, { useState } from "react";
 import Button from "react-bootstrap/Button";
 import Card from "react-bootstrap/Card";
 import AuthService from "../services/auth.service";
-import axios from 'axios'
+import axios from "axios";
+import React, { useEffect, useRef } from "react";
+import { useState } from "react";
 
-function Form(props) {
+function Form() {
   var numIngredients = 0;
   var numSteps = 0;
-
+  const cloudinaryRef = useRef();
+  const widgetRef = useRef();
+  const [loading, setLoading] = useState(false);
+  const [url, setUrl] = useState([]);
   const [recipe, setRecipe] = useState({
     title: "",
     blurb: "",
     servings: "",
-    cost: {
-      total: "",
-      perServing: "",
-    },
-
-    img: "",
+    urlSource: "",
     totalTime: {
-      cookTime: "",
-      activeTime: "",
+      cook: "",
+      active: "",
     },
     ingredients: [],
     steps: [],
-    user: ""
+    user: "",
   });
 
+  useEffect(() => {
+    cloudinaryRef.current = window.cloudinary;
+    widgetRef.current = cloudinaryRef.current.createUploadWidget(
+      {
+        cloudName: "dype2qwpo",
```

Comment:
Should this be an environment variable .env

Reply:
I'll try to add this change